app/components/channel_drawer_list/channel_drawer_list.js

```diff
@@ -172,9 +172,10 @@ class ChannelDrawerList extends Component {
             currentTeam
         } = this.props;
 
-        this.props.onSelectChannel(channel.id);
-        this.props.actions.viewChannel(currentTeam.id, channel.id);
-        this.props.actions.markChannelAsRead(channel.id, currentChannel.id);
+        this.props.onSelectChannel(channel.id).then(() => {
```

Comment:
maybe make the function async and instead of using `then` we can do an `await` here